### DISCUSSION TITLE
System Bridge on Proxmox - Battery data

### DISCUSSION THREAD

**Discussion by cristiancporto** [timmo001/system-bridge#3313] [Help]
I have Proxmox (Proxmox VE 8.2.7) on a laptop that works as my home server, and on Proxmox I have HomeAssistant. So, I wanted to log in HA the battery status, and charging state, so I can look in the future how many times there were power outages and see the battery usage.
Then, I went to System Bridge as it looked like the tool to solve this, as it has HA integration.
After installing the .deb in proxmox, I added the integration on HA but it does not show battery data.
I was able to test by HTTP requests to System Bridge, and when I send a GET to 
http://[IP]:[PORT]/api/data/battery, the response is
```json
{
    "is_charging": null,
    "percentage": null,
    "time_remaining": -1
}
```
Before installing System Bridge on Proxmox, I installed upower and lshw as asked in the intallation guide, so it doesn't seem to be this the problem.

Also, running `upower -e` lists:
```
/org/freedesktop/UPower/devices/battery_BAT1
/org/freedesktop/UPower/devices/line_power_ADP1
/org/freedesktop/UPower/devices/DisplayDevice
```
and with `upower -i /org/freedesktop/UPower/devices/battery_BAT1`:
```
native-path:          BAT1
  vendor:               SAMSUNG Electronics
  power supply:         yes
  updated:              Tue 05 Nov 2024 02:06:18 PM -03 (15 seconds ago)
  has history:          yes
  has statistics:       yes
  battery
    present:             yes
    rechargeable:        yes
    state:               fully-charged
    warning-level:       none
    energy:              38.073 Wh
    energy-empty:        0 Wh
    energy-full:         38.85 Wh
    energy-full-design:  48.84 Wh
    energy-rate:         0 W
    voltage:             12.165 V
    charge-cycles:       607
    percentage:          98%
    capacity:            79.5455%
    technology:          lithium-ion
    icon-name:          'battery-full-charged-symbolic'
```
and `upower -i /org/freedesktop/UPower/devices/line_power_ADP1`:
```
  native-path:          ADP1
  power supply:         yes
  updated:              Tue 05 Nov 2024 09:33:48 AM -03 (16485 seconds ago)
  has history:          no
  has statistics:       no
  line-power
    warning-level:       none
    online:              yes
    icon-name:          'ac-adapter-symbolic'
```
It seems upower can see the data, but System Bridge isn't able to get it?

**Comment by raptr569** [comment]
I have the exact same issue with my Debian 12 tablet.

upower shows all the expected battery information and stats but system-bridge is either not picking up the information or not passing it to the API for home assistant to collect.

**Reply by raptr569** [reply]
@timmo001 - would you be able to elaborate on the mechanics used to collect the battery information?

**Reply by toeachbyeach** [reply]
Having the same issue here with Ubuntu 20.04. Hopefully @timmo001 can help us out!
